### PR TITLE
Hai 571 update s3 permissions

### PR DESCRIPTION
**Pull request by comozo**
Resolves https://github.com/harrison-ai/platform/issues/86

## What

- Set bucket ownership to BucketOwnerEnforced
- Update TF version to the latest version.

## Why

We want the bucket objects ACL to be by default own by the AWS account owner so the account owner can freely manage the objects

## Notes
This section is also optional and should include anything else that you would like to discuss in the PR review that is not captured elsewhere.

* See the [data engineering handbook](https://dataengineering.harrisonai.io/pages/source_control.html#anatomy-of-a-pull-request) for more info.

### CODE REVIEW

**Comment by comozo** [comment]
Result of the change on my test bucket

image

File: main.tf [hunk 1]
```diff
@@ -34,6 +34,14 @@ resource "aws_s3_bucket_versioning" "this" {
...
 }
 
+resource "aws_s3_bucket_ownership_controls" "this" {
+  bucket = aws_s3_bucket.this.id
+
+  rule {
+    object_ownership = "BucketOwnerEnforced"
+  }
+}
+
```
Comment (schattingh) [inline main.tf:44]: I notice the official S3 module has an explicit depends on statement.  I wonder if we should add it here too?  https://github.com/terraform-aws-modules/terraform-aws-s3-bucket/blob/master/main.tf#L747

**Review by schattingh — CHANGES_REQUESTED** [review]
Just made one suggested addition.  Also, we'll cut a new release from this, so can you update the changelog too please?

File: README.md [hunk 2]
```diff
@@ -42,7 +41,7 @@ module "this" {
 
 | Name | Version |
 |------|---------|
-| <a name="provider_aws"></a> [aws](#provider\_aws) | >= 4.0.0 |
+| <a name="provider_aws"></a> [aws](#provider\_aws) | 4.58.0 |
```
Comment (schattingh) [inline README.md:44]: Has there perhaps been an uncommitted change to the provider to introduce this change to the provider version?   This should be reverted as we can't force the latest minor version I don't think.

**Review by dcarrion87 — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (dcarrion87) [reply README.md]: I think this should just be ">= 4.0.0" ?

Is the auto doc generator blasting it away?

**Review by dcarrion87 — APPROVED** [review]

**Review by schattingh — APPROVED** [review]
nice @comozo, thanks

**Pull request merged**